### DISCUSSION TITLE
HEAD is now at 805f0ba3c

### DISCUSSION THREAD

**Discussion by jacobhroutzong** [Homebrew/discussions#1802] [Getting started]
Merge pull request #11666 from carlocab/optim-block

Just tried installing Homebrew again after significant interruptions due to connection/network. Anyone know if this is an error or am I doing this correctly?

**Comment by jacobhroutzong** [comment]
I'm not even 100% sure i'm doing this right. I've tried installing it several times and am still unsure if it successfully downloaded.

**Comment by carlocab** [comment]
Trying checking #666 for related problems and solutions. It's impossible to say what happened without the exact output and error message.

**Reply by jacobhroutzong** [reply]
I'm installing on m1 mac. Is there anyway to test if I've done this correctly?

**Reply by carlocab** [reply]
Yes. Do `brew doctor`. It should tell you `Your system is ready to brew`.

**Reply by gromgit** [reply]
And if it doesn't, do whatever `brew doctor` tells you to fix the situation.

**Reply by jacobhroutzong** [reply]
My problems are probably more closely aligned with what was going on in #1719. I'm just unsure of how to follow these instructions:
 Add Homebrew to your PATH in ~/.zprofile:
 echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
 eval $(/opt/homebrew/bin/brew shellenv)
Or how to even get them to appear again.

**Reply by jacobhroutzong** [reply]
> Yes. Do `brew doctor`. It should tell you `Your system is ready to brew`.

I did brew doctor and it says, "zsh: command not found: brew"

**Reply by gromgit** [reply]
> My problems are probably more closely aligned with what was going on in #1719. I'm just unsure of how to follow these instructions:
> Add Homebrew to your PATH in ~/.zprofile:
> echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
> eval $(/opt/homebrew/bin/brew shellenv)
> Or how to even get them to appear again.

Those are the exact commands you need to type (though it's better to copy-and-paste for accuracy) into the Terminal. This:
```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
```
add the necessary instructions for macOS to make `brew` commands available to all Terminals upon login, while:
```
eval $(/opt/homebrew/bin/brew shellenv)
```
makes `brew` commands available to your _current_ Terminal.

So to get the necessary diagnostic output, you'd need to do this:
```
eval $(/opt/homebrew/bin/brew shellenv)
brew config
brew doctor
```

**Reply by jacobhroutzong** [reply]
That seemed to do the trick. brew doctor
Your system is ready to brew.
